npm/webpack-batteries-included-preprocessor/index.js

```diff
@@ -131,23 +132,51 @@ const getDefaultWebpackOptions = () => {
         loader: require.resolve('coffee-loader'),
       }],
     },
+    plugins: [
+      new webpack.ProvidePlugin({
+        Buffer: ['buffer', 'Buffer'],
+        process: 'process/browser',
+      }),
+    ],
     resolve: {
       extensions: ['.js', '.json', '.jsx', '.mjs', '.coffee'],
-      alias: {
-        'child_process': require.resolve('./empty'),
-        'cluster': require.resolve('./empty'),
-        'console': require.resolve('./empty'),
-        'dgram': require.resolve('./empty'),
-        'dns': require.resolve('./empty'),
-        'fs': require.resolve('./empty'),
-        'http2': require.resolve('./empty'),
-        'inspector': require.resolve('./empty'),
-        'module': require.resolve('./empty'),
-        'net': require.resolve('./empty'),
-        'perf_hooks': require.resolve('./empty'),
-        'readline': require.resolve('./empty'),
-        'repl': require.resolve('./empty'),
-        'tls': require.resolve('./empty'),
+      fallback: {
```

Comment:
(unrelated to this line of code, general comment) This should give us out of the box support for things like 

- optional chaining `?.`
- class properties:

```js
class Foo {
  bar = '123'
}
```

Maybe we can add a few tests to make sure this is the case?

Related:

- #22788
- #25300

Reply:
I can take a look into this and see if I can add a few tests

Reply:
@lmiller1990 I took a look into both of these. The first looks like its an issue with not transpiling `node_modules`? And I couldnt reproduce the later with Webpack 4. Did you have any luck?

Reply:
@lmiller1990 I took a look into both of these. The first looks like its an issue with not transpiling `node_modules`? And I couldnt reproduce the later with Webpack 4. Did you have any luck?

Reply:
Hm - looks like those issues are not related to webpack 5. Disregard this.